### DISCUSSION TITLE
Jobs for no reason stuck in pending status

### DISCUSSION THREAD

**Discussion by klinux** [terrakube-io/terrakube#714] [Q&A] [answered]
I have workspaces that stuck in pending status.

The api server has this message in the logs:
`Job 348 is waiting for previous jobs to be completed`

Theres no others jobs running or pending before 348:

![Screenshot_1172](https://github.com/AzBuilder/terrakube/assets/941051/24351b15-e176-4ed8-9179-0043348f94a7)
![Screenshot_1171](https://github.com/AzBuilder/terrakube/assets/941051/f008b996-86e4-4a42-b1fe-946fa52778be)

Theres some caches or others things that can lock the execution jobs?

Thank you

**Comment by alfespa17** [comment]
Hello @klinux, that one is a bug in version 2.19.0, you can check this [comment](https://github.com/orgs/AzBuilder/discussions/712#discussioncomment-8479419)

We foget to add the status cancelled in this validation  

https://github.com/AzBuilder/terrakube/blob/191689a34124232295816fad2de38294ffd71075/api/src/main/java/org/terrakube/api/plugin/scheduler/ScheduleJob.java#L86

As a temporary fix you could go to the database and check if you have some status as cancelled for a particular workspace and change that to failed, I will release a hotfix later today

**Reply by klinux** [reply]
Thank you @alfespa17

**Reply by klinux** [reply]
@alfespa17
I think that I have other problem here, I updated all jobs em steps to failed instead cancelled, but the stuck job still here, can I look to other thing?

**Reply by alfespa17** [reply, marked as answer]
Check all the previous jobs from a particualr workspace and check the status to see if all are in faild/completed or any other state

**Reply by klinux** [reply]
You 're right, theres some columns as rejected status. The workaround works.

**Reply by alfespa17** [reply]
The job status issue was fixed in [2.19.1](https://github.com/AzBuilder/terrakube/releases/tag/2.19.1), the release is running the container images will be ready in a few minutes
